src/components/ArticleDetail.jsx

```diff
@@ -18,6 +19,15 @@ function ArticleDetail() {
       });
     }, []);
   
+  function handleVote(event) {
+    const incVote = voted ? -1 : 1;
+    setVoted((current) => !current);
+    setArticle((current) => {
+      return {...current, votes: current.votes + incVote};
+    });
+    voteArticle(article_id, incVote);
```

Comment:
What happens if this doesnt update successfully? needs to be reflected in the FE.
Could you set the votes back to what they were and provide an error message?

Reply:
At the  moment, if you click the button whilst offline, its looks like its updated, but on the database, it hasnt.